src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -2312,6 +2316,76 @@ public URL createCrumbedUrl(String relativePath) throws IOException {
         }
 
         /**
+         * Add the "Authorization" header with Basic credentials derived from login and password using Base64
+         * @since TODO
+         */
+        public @NonNull WebClient usingBasicCredentials(@NonNull String login, @NonNull String passwordOrToken){
```

Comment:
What is the reason of having both "using..." and "with..."?

Reply:
New design proposed, with an optional closure as last parameter. This will avoid the confusion between the `AutoCloseable` (that remove the header) and the `WebClient` (that is already `AutoCloseable`).
Additionnally it will simplify the readability of the tests.